From: Manish Dusane <manish.dusane@qlogic.com>
Date: Thu, 20 Sep 2012 07:35:11 -0400
Subject: [PATCH 60/63] qla4xxx: Fix double clearing of risc_intr for ISP83XX
Git-commit: 48a859d29f3d7a855093819633abcc9c877108ef
Patch-mainline: v3.7-rc1
References: FATE#313902

Issue:
For ISP83XX risc_intr getting clear two times in case of polling mode.
risc_intr getting clear from interrupt_service_routine() as well
from process_mbox_intr() because of this driver may clear risc_intr
without processing interrupt.

Fix:
Do not clear risc_intr from process_mbox_intr().

Signed-off-by: Manish Dusane <manish.dusane@qlogic.com>
Signed-off-by: Vikas Chaudhary <vikas.chaudhary@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Acked-by: Lee Duncan <lduncan@suse.com>
---
 drivers/scsi/qla4xxx/ql4_83xx.c | 1 -
 1 file changed, 1 deletion(-)

 drivers/scsi/qla4xxx/ql4_83xx.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/scsi/qla4xxx/ql4_83xx.c b/drivers/scsi/qla4xxx/ql4_83xx.c
index f963b06..8b367ef 100644
--- a/drivers/scsi/qla4xxx/ql4_83xx.c
+++ b/drivers/scsi/qla4xxx/ql4_83xx.c
@@ -1402,7 +1402,6 @@ void qla4_83xx_process_mbox_intr(struct scsi_qla_host *ha, int outcount)
 		ha->mbox_status_count = outcount;
 		ha->isp_ops->interrupt_service_routine(ha, intr_status);
 	}
-	writel(0, &ha->qla4_83xx_reg->risc_intr);
 }
 
 /**
-- 
1.8.0


